### ISSUE TITLE
Admin Console download of 5.5GB Ubongo Videos won't start, or is stuck, and Kiwix catalog cannot be refreshed? [Later on, all ZIM files appear as "502 Bad Gateway"]

### ISSUE THREAD

**Issue by holta** [iiab/iiab#3812]
Trying to download the 5.5GB Ubongo Videos (full set) leads to "SCHEDULED" (screenshot further below) but it never starts:

![Upload 2024-09-29 193649](https://github.com/user-attachments/assets/5191aa70-782b-4ffb-951b-6c85f3a3ec1b)

![Install Content 2024-09-29 193735](https://github.com/user-attachments/assets/ef050c97-e7dc-47b0-a818-1d78a7efe1ba)

Canceling individual jobs ("Cancel Checked Jobs") was attempted and did not work (and really should 😉).

At a deeper level, https://download.kiwix.org/zim/videos/ does not show the above `ubongo_sw_playlist-*` files being available anymore, likely due to Kiwix housekeeping that began August 8th:

- openzim/zim-requests#1132

"Refresh Kiwix Catalog" was attempted (repeatedly) and is not working: 

![image (1)](https://github.com/user-attachments/assets/c44e22a6-9659-45dd-a80a-832f4a4f2615)

It sounds like they've rebooted and tried a lot of different things to get unstuck, without success.

Any suggestions to help them get going again?  (I'm not sure if they have command-line experience, but if necessary I could ask them to delete specific-or-stuck files &/or to try to run things like `sudo iiab-get-kiwix-cat` more forcefully, etc?)

**Comment by tim-moody** [comment]
Please verify that Kiwix is installed

**Comment by tim-moody** [comment]
Had a chance to look at this some more and I see that indeed Kiwix looks to be installed.

First, for me get kiwix catalog works.

There's no doubt that scheduling a download of a zim that is then yanked out of the catalog and removed from the server could easily cause problems, but looking at the code I don't see why it would not be able to be cancelled, especially if it hadn't started. If there was a reboot that moved it from scheduled to restarted there might be a problem, but I don't see how that happened either.

I also note that the 2024 Ubongo is much larger than 2023 so not sure if that broke anything.

Hard to diagnose without access to the server.

**Comment by holta** [comment]
He [person asking the original question] replied:

> I seem to be going backward.  Now, I get a 502 Bad Gateway error for most of the content I've downloaded.  Maps work, great books work, Khan Academy works, etc.   Ubongo, most of Wiki (except Medical which works), English Wiki, and most everything else I get 502 Bad Gateway error.  I've tried rebooting several times with no success.  Hmmm.  Any suggestions?

> BTW, I did rerun the re-index KWIX and it was successful.

PS I've asked him to try to run `sudo iiab-diagnostics` from the IIAB's Linux command-line.

**Comment by tim-moody** [comment]
>Now, I get a 502 Bad Gateway error for most of the content I've downloaded.

It seems this actually only refers to ZIMs, Kiwix content. There could be a mismatch between the ZIMs and your version of kiwix or something out of sync

I would start with Install Content -> Reindex Kwix Content, which it seems you did

then check the version of kiwix in iiab-diagnostics

**Comment by holta** [comment]
He was able to log in to the IIAB (RPi) over ssh, but the iiab-diagnostics appear too large to easily extract.  I'll ask if he can try https://temp.sh which sometimes works, much like a pastebin, but for larger files.

**Comment by holta** [comment]
Tangentially here's a kiwix-serve bug fix to watch for soon, in upcoming or nightly releases:

- kiwix/libkiwix#1132

**Comment by tim-moody** [comment]
running /opt/iiab/kiwix/bin/kiwix-serve --version might give us some info

ls -lav /library/zims/content/ would tell us what zims

**Comment by holta** [comment]
kiwix-serve is unable to start, apparently due to a problem with `/library/zims/library.xml` :

![image3](https://github.com/user-attachments/assets/e9380663-79c4-4e4a-9cef-7b33dfc4271a)

**Comment by holta** [comment]
Does the failure of `ls -lav /library/zims/content` mean something potentially more serious, like a disk issue ??

**Comment by tim-moody** [comment]
a lot of this still makes me wonder if kiwix installed completely

**Comment by holta** [comment]
> a lot of this still makes me wonder if kiwix installed completely

Yes, as he's apparently trying to add more content to an IIAB that was originally installed in June 2023.
